### PR TITLE
fix(regulations-admin): Update admin backend client

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

* Change `RegulationsAdminClientService` from `RESTDataSource`.
* OpenAPI and generated client with Enhanced Fetch.
* Minor regulation-admin home screen change, display error message and empty screen message.

## Why

* OpenAPI + Enhanced Fetch is the preferred way.
* Home screen change: Improve information flow to the user.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thorkellmani — APPROVED** [review]
lgtm

**Review by brynjarorng — APPROVED** [review]
LGTM

File: libs/clients/regulations-admin/project.json
Comment (saevarma) [inline libs/clients/regulations-admin/project.json:21]: This task should be named:
```suggestion
    "schemas/openapi-generator": {
```

File: libs/clients/regulations-admin/README.md
Comment (saevarma) [inline libs/clients/regulations-admin/README.md:1]: I guess that this client library is also missing codeowner configuration.

File: libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts
Comment (saevarma) [inline libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts:97]: You should be able to use the return type of the `draftRegulationControllerCreate` instead of `any` to define a proper return type.

File: libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts
Comment (saevarma) [inline libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts:62]:
```suggestion
    return this.draftRegulationsApiWithAuth(
      auth,
    ).draftRegulationControllerGetAll({
      page: page ?? 0,
    })
```

File: libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts
Comment (saevarma) [inline libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts:66]: Unnecessary `await` and then the `async` can be removed from the function definition. Should just simplify this for all functions in this service that are doing `return await ...`
```suggestion
    return this.draftRegulationsApiWithAuth(
```

File: libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts
Comment (saevarma) [inline libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts:111]: Ditto on `any` return type.

File: libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts
Comment (saevarma) [inline libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts:187]: Ditto on `any` return type.

File: libs/clients/regulations-admin/src/lib/apiConfiguration.ts
Comment (saevarma) [inline libs/clients/regulations-admin/src/lib/apiConfiguration.ts:43]: This can be simplified by doing

```
export const exportedApis = [DraftAuthorApi, DraftRegulationCancelApi, DraftRegulationChangeApi, DraftRegulationsApi].map(
  (Api) => ({
    provide: Api,
    useFactory: (configuration: Configuration) => new Api(configuration),
    inject: [ApiConfiguration.provide],
  }),
)
```

Where the configuration is created as a provider

```
export const ApiConfiguration = {
  provide: 'RegulationsAdminClientConfiguration',
  useFactory: (config: ConfigType<typeof RegulationsAdminClientConfig>) => {
    return new Configuration({
      fetchApi: createEnhancedFetch({
        name: 'clients-regulations-admin-api',
      }),
      basePath: config.baseApiUrl,
       headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
    })
  },
  inject: [RegulationsAdminClientConfig.KEY],
}
```
You can see an example here: https://github.com/island-is/island.is/blob/main/libs/clients/auth/delegation-api/src/lib/apis.ts

**Review by saevarma — APPROVED** [review]
Overall LGTM.
Few pointers that can be improved before this is merged.

**Pull request merged**
